Content.Server/Chat/Systems/LastMessageBeforeDeathSystem.cs

```diff
@@ -0,0 +1,189 @@
+using System.Text;
+using Content.Server.Discord;
+using Content.Shared.Mobs.Systems;
+using System.Collections.Generic;
+using Content.Server.GameTicking;
+using Content.Server.Database;
+using Robust.Shared.Network;
+using Robust.Shared.Player;
+using Content.Server.Discord.Managers;
+using Robust.Shared.Configuration;
+using System.Threading.Tasks;
+using Content.Shared.CCVar;
+using Robust.Shared.Enums;
+
+namespace Content.Server.Chat.Systems
+{
+    internal class LastMessageBeforeDeathSystem : EntitySystem
+    {
+        [Dependency] private readonly IServerDbManager _db = default!;
+        [Dependency] private readonly IConfigurationManager _configManager = default!;
+        [Dependency] private readonly IEntityManager _entityManager = default!;
+        [Dependency] private readonly ILogManager _logManager = default!;
+        [Dependency] private readonly DiscordWebhook _discord = default!;
+
+        private MobStateSystem _mobStateSystem = default!;
+        private GameTicker _gameTicker = default!;
+        private IEntityManager _entManager = default!;
+
+        private ISawmill _sawmill = default!;
+
+        private readonly LastMessageWebhookManager _webhookManager = new LastMessageWebhookManager();
+        private Dictionary<ICommonSession, PlayerData> _playerData = new Dictionary<ICommonSession, PlayerData>();
+
+        private const string MaxICLengthCVar = "chat.max_ic_length";
+        private const int MaxWebhookMessageLength = 2000; // This should not really be changed... since the maximum characters per message for webhooks is 2000.
+
+        private static readonly Random _random = new Random();
+
+        private WebhookIdentifier? _webhookIdentifierLastMessage;
+
+        private class PlayerData
+        {
+            public Dictionary<string, CharacterData> Characters { get; set; } = new Dictionary<string, CharacterData>();
```

Comment:
This never needs to be modified, so you can do this.

```suggestion
            public Dictionary<string, CharacterData> Characters { get; } = new Dictionary<string, CharacterData>();
```